chat-components/src/services/BroadcastService.ts

```diff
@@ -45,7 +104,16 @@ export const BroadcastService = {
          * Omit copying methods to prevent 'DataCloneError' in older browsers when passing an object with functions
          * This exception occurs when an object can't be clone with the 'structured clone algorithm' (used by postMessage)
          */
-        pubChannel.postMessage(JSON.parse(JSON.stringify(message)));
+        try {
+            const messageCopy = JSON.parse(JSON.stringify(message));
+            const eventId = new Date().getTime();
```

Comment:
lets evaluate structures that allow you to use id as index, including keeping order such , https://www.npmjs.com/package/immutable